python/lsst/cbp/beamInfo.py

```diff
@@ -0,0 +1,136 @@
+        Name of beam
+    holePos : pair of `float`
+        Position of hole on CBP mask (x, y mm)
+    isOnPupil : `bool`
+        Beam is likely on the pupil and not obscured by the secondary
+    isOnFocalPlane : `bool`
+        Beam is likely on the focal plane;
+        this is independent of ``isOnPupil``; both must be true
+        for the beam to be detected
+    isOnDetector : `bool`
+        Beam is probably on a detector; this is unreliable
+    focalPlanePos : `lsst.geom.Point2D`
+        Telescope :ref:`focal plane <lsst.cbp.focal_plane>` position
+        of beam (x, y mm)
+    focalFieldAngle : `lsst.geom.Point2D`
+        :ref:`Focal plane field angle <lsst.cbp.focal_plane_field_angle>`
+        of beam (x, y rad)
+    pupilFieldAngle : `lsst.geom.Point2D`
+        :ref:`Pupil field angle <lsst.cbp.pupil_field_angle>`
+        of beam (x, y rad)
+    pupilPos : `lsst.geom.Point2D`
+        Telescope :ref:`pupil plane position <lsst.cbp.pupil_position>`
+        of beam (x, y mm)
+
+    """
+    def __init__(self, cameraGeom, name, holePos, isOnPupil, isOnFocalPlane, focalPlanePos, pupilPos,
+                 focalFieldAngle, pupilFieldAngle):
+        self._cameraGeom = cameraGeom
+        self.name = name
+        self.holePos = holePos
+        self.isOnPupil = isOnPupil
+        self.isOnFocalPlane = isOnFocalPlane
+        self.focalPlanePos = Point2D(*focalPlanePos)
+        self.focalFieldAngle = Point2D(*focalFieldAngle)
+        self.pupilFieldAngle = Point2D(*pupilFieldAngle)
+        self.pupilPos = Point2D(*pupilPos)
+        fieldAngleToFocalPlane = self._cameraGeom.getTransform(FIELD_ANGLE, FOCAL_PLANE)
+        self.focalPlanePos = fieldAngleToFocalPlane.applyForward(self.focalFieldAngle)
+        self._isOnDetector = None
+        self._detector = None
+
+    @property
+    def isOnDetector(self):
+        """True if the beam is on a detector
+        """
+        if self._isOnDetector is None:
+            self._findDetector()
+        return self._isOnDetector
+
+    @property
+    def detectorPos(self):
+        """The position of the spot on the detector,
+        or (nan, nan) if isOnDetector is False
+        """
+        if not self.isOnDetector:
+            return Point2D(np.nan, np.nan)
+        pixelSys = self._detector.makeCameraSys(PIXELS)
+        focalPlaneToPixels = self._cameraGeom.getTransform(FOCAL_PLANE, pixelSys)
+        return focalPlaneToPixels.applyForward(self.focalPlanePos)
+
+    @property
+    def detectorName(self):
+        """The name of the detector that the beam falls on,
+        or None if isOnDetector is False
+        """
+        if not self.isOnDetector:
+            return None
+        return self._detector.getName()
+
+    def _findDetector(self):
+        detectors = self._cameraGeom.findDetectors(self.focalFieldAngle, FIELD_ANGLE)
+        if not detectors:
+            self._isOnDetector = False
+        else:
+            self._detector = detectors[0]
```

Comment:
In theory it is possible for a camera to have overlapping detectors. In practice I don't expect that for this usage so it's returning any one of the found detectors. I could enforce that, if you think it preferable.

Reply:
I think this is probably fine as-is then. I actually wasn't thinking about the case of overlapping detectors (how does that even work?!), it was more me imagining the case where the spots are modeled as having finite size, and so land on more than one detector. However, if you feel like changing it to raising an exception I don't think that could hurt, and might be nice as it would show if weird things are happening in the camera geometry and being swept under the rug...